packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/dynamoDb-walkthrough.ts

```diff
@@ -644,137 +666,142 @@ async function addTrigger(context: $TSContext, resourceName: string, triggerList
   const resourceDirPath = path.join(projectPath, 'amplify', 'backend', AmplifyCategories.STORAGE, resourceName);
   const cfnFilePath = path.join(resourceDirPath, `${resourceName}-cloudformation-template.json`);
 
   // Removes dangling commas from a JSON
-  const removeDanglingCommas = (value: any) => {
+  const removeDanglingCommas = (value: string): string => {
     const regex = /,(?!\s*?[{["'\w])/g;
 
     return value.replace(regex, '');
   };
 
-  /* Current Dynamo CFN's have a trailing comma (accepted by CFN),
-  but fails on JSON.parse(), hence removing it */
+  /* Current Dynamo CFN's have a trailing comma (accepted by CFN), but fails on JSON.parse(), hence removing it */
 
-  let oldcfnString = fs.readFileSync(cfnFilePath, 'utf8');
-  oldcfnString = removeDanglingCommas(oldcfnString);
+  let oldCfnString = fs.readFileSync(cfnFilePath, 'utf8');
+  oldCfnString = removeDanglingCommas(oldCfnString);
 
-  const oldCfn = JSON.parse(oldcfnString);
+  const oldCfn = JSON.parse(oldCfnString);
 
-  const newCfn = {};
+  const newCfn: $TSAny = {};
 
   Object.assign(newCfn, oldCfn);
 
   // Add env parameter
-  if (!(newCfn as any).Parameters) {
-    (newCfn as any).Parameters = {};
+  if (!newCfn.Parameters) {
+    newCfn.Parameters = {};
   }
 
-  (newCfn as any).Parameters.env = {
+  newCfn.Parameters.env = {
     Type: 'String',
   };
 
   // Add conditions block
-  if (!(newCfn as any).Conditions) {
-    (newCfn as any).Conditions = {};
+  if (!newCfn.Conditions) {
+    newCfn.Conditions = {};
   }
 
-  (newCfn as any).Conditions.ShouldNotCreateEnvResources = {
+  newCfn.Conditions.ShouldNotCreateEnvResources = {
     'Fn::Equals': [
-      {
-        Ref: 'env',
-      },
+      { Ref: 'env' },
       'NONE',
     ],
   };
 
   // Add if condition for resource name change
-  (newCfn as any).Resources.DynamoDBTable.Properties.TableName = {
+  newCfn.Resources.DynamoDBTable.Properties.TableName = {
     'Fn::If': [
       'ShouldNotCreateEnvResources',
-      {
-        Ref: 'tableName',
-      },
+      { Ref: 'tableName' },
       {
         'Fn::Join': [
           '',
           [
-            {
-              Ref: 'tableName',
-            },
+            { Ref: 'tableName' },
             '-',
-            {
-              Ref: 'env',
-            },
+            { Ref: 'env' },
           ],
         ],
       },
     ],
   };
 
   const jsonString = JSON.stringify(newCfn, null, '\t');
-
   fs.writeFileSync(cfnFilePath, jsonString, 'utf8');
-}
-
-export function getIAMPolicies(resourceName: string, crudOptions: $TSAny) {
-  let policy = {};
+};
+
+/**
+ * getIAMPolicies: get the IAM policies for CRUD
+ * @param resourceName - The resourceName
+ * @param crudOptions - list of CRUD actions
+ * @returns DynamoDBCLIInputs
+ */
+export const getIAMPolicies = (resourceName: ResourceName, crudOptions: $TSAny): { policy: $TSObject, attributes: string[] } => {
```

Comment:
For future me or other reviewers, the functional changes in this PR are adding the PartiQL* actions to the policies here